### PR TITLE
refactor: process federation connection removed event - WPB-10187

### PR DESCRIPTION
**Pull request by jullianm**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10187" title="WPB-10187" target="_blank">TaskWPB-10187</a>  [iOS] Process FederationConnectionRemovedEvent
  </td></table>
  <br />
 

### Key points

This PR is part of the quick sync refactoring plan and is related to processing the multiple events we receive from the backend or the push channel.

Specifically, this PR is about porting the existing implementation of the `FederationConnectionRemovedEventProcessor` event.

Legacy code was using a FederationTerminationManager part of WireRequestStrategy framework to manage the storage logic. 

```
 federationTerminationManager.handleFederationTerminationBetween(firstDomain, otherDomain: secondDomain)
```

To avoid using this internal dependency, we now directly perform the storage logic operations in our dedicated ConnectionRepository object.

### Testing

UTs cover the following use cases:
- Federated participants are removed from the conversations that are not hosted on the given domains.
- A federated participant on a hosted conversation domain is removed from that conversation if he's not part of the same domain.

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireDomain/Sources/WireDomain/Repositories/ConnectionsRepositiory.swift [hunk 1]
```diff
@@ -30,13 +30,24 @@ protocol ConnectionsRepositoryProtocol {
     /// Pull self team metadata frmo the server and store locally.
 
     func pullConnections() async throws
+
+    /// Removes a federation connection between two domains.
+    ///
+    /// - Parameter domain : The domain for which the connection was removed.
```
Comment (netbe) [inline WireDomain/Sources/WireDomain/Repositories/ConnectionsRepositiory.swift]:
```suggestion
    /// - Parameter domain: The domain for which the connection was removed.
```

File: WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    var context: NSManagedObjectContext {
+        coreDataStack.syncContext
+    }
+
+    override func setUp() async throws {
+        coreDataStack = try await coreDataStackHelper.createStack()
+        sut = FederationConnectionRemovedEventProcessor(
+            repository: ConnectionsRepository(connectionsAPI: mockAPI, context: context)
+        )
+        try await super.setUp()
```
Comment (netbe) [inline WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift]:
```suggestion
        try await super.setUp()
        coreDataStack = try await coreDataStackHelper.createStack()
        sut = FederationConnectionRemovedEventProcessor(
            repository: ConnectionsRepository(connectionsAPI: mockAPI, context: context)
        )
```
question: any reason for doing the setup last?

File: WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    override func tearDown() async throws {
+        coreDataStack = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+        try await super.tearDown()
```
Comment (netbe) [inline WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift]: same here

**Review by netbe — APPROVED** [review]
LGTM:) just added minor comments

File: WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift (hunk 2 above)
Comment (jullianm) [reply WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift]: fixed thx

File: WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift (hunk 3 above)
Comment (jullianm) [reply WireDomain/Tests/WireDomainTests/Event Processor/FederationEventProcessor/FederationConnectionRemovedEventProcessorTests.swift]: fixed thx

**Comment by jullianm** [comment]
closing this PR, code will be part of of the following one [here](https://github.com/wireapp/wire-ios/pull/1918)

**Pull request closed without merging**